paimon-core/src/main/java/org/apache/paimon/table/Table.java

```diff
@@ -79,6 +79,9 @@ public interface Table extends Serializable {
     @Experimental
     void rollbackTo(String tagName);
 
+    /** Record the consumer of the table through consumer id and next snapshot id. */
+    void recordConsumer(String consumerId, long nextSnapshotId);
```

Comment:
Don't introduce public API now. It is not so stable.

Reply:
done.